libarchive/archive_read_support_format_tar.c

```diff
@@ -2487,7 +2487,7 @@ pax_attribute(struct archive_read *a, struct tar *tar, struct archive_entry *ent
 			}
 			else if (key_length == 5 && memcmp(key, "nlink", 5) == 0) {
 				if ((err = pax_attribute_read_number(a, value_length, &t)) == ARCHIVE_OK) {
-					archive_entry_set_nlink(entry, t);
+					archive_entry_set_nlink(entry, (unsigned int)t);
```

Comment:
I think there's an actual bug here:  `pax_attribute_read_number` will parse negative values, and `nlink` should never be negative.  So we should probably extend the `if` condition here to require that `t > 0`.

Reply:
The goal was just to fix warnings (no runtime changes except I guess not with ubsan), but I can go ahead and make this change tomorrow

Reply:
In the event that `t` _is_ negative, I'm guessing we should set the error string & return failure (`ARCHIVE_ERROR`? maybe `ARCHIVE_WARN`?), but I'm not familiar enough with this code path to know what best to use/say if you have suggestions